mm2src/lp_network.rs

```diff
@@ -226,6 +228,25 @@ pub async fn request_relays<T: de::DeserializeOwned>(
     Ok(parse_peers_responses(responses))
 }
 
+pub async fn request_addresses<T: de::DeserializeOwned>(
+    ctx: MmArc,
+    req: P2PRequest,
+    peers_addresses: Vec<(String, String)>,
+) -> Result<Vec<(PeerId, PeerDecodedResponse<T>)>, String> {
+    let encoded = try_s!(encode_message(&req));
```

Comment:
Could you please switch this to MmError with the question mark operator? Please consider the `try_s!` approach as deprecated.